### PR TITLE
chore: increment go mod to v6

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Updating `go.mod` to `ibc-go/v6`

closes: #2011 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by chatton — APPROVED** [review]
LGTM once all the tests are passing 🥇

**Review by crodriguezvega — APPROVED** [review]
Awesome!

File: modules/apps/27-interchain-accounts/host/keeper/account_test.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package keeper_test
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/host/keeper/account_test.go:1]: Is this from a merge conflict?

Comment (damiannolan) [reply modules/apps/27-interchain-accounts/host/keeper/account_test.go:1]: do you mean the entire test file?

Comment (colin-axner) [reply modules/apps/27-interchain-accounts/host/keeper/account_test.go:1]: Yea. Why was the test file added for a version bump pr?

Comment (damiannolan) [reply modules/apps/27-interchain-accounts/host/keeper/account_test.go:1]: Looks like I cut this branch before the PR which removed this. So when I pulled in the latest changes this file was kept.

I'll push an update to remove now. Good catch! :)

**Review by colin-axner — APPROVED** [review]

**Pull request merged**